File: CS/13_Conversion/ToImage.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;

namespace ToImage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the path of the PDF file to be processed.
            String file = @"..\..\..\..\..\..\Data\ToImage.pdf";

            // Create a new PdfDocument object.
            PdfDocument doc = new PdfDocument();

            // Load the PDF document from the specified file.
            doc.LoadFromFile(file);

            // Save each page of the PDF document as an image.
            for (int i = 0; i < doc.Pages.Count; i++)
            {
                // Generate a unique file name for each image.
                String fileName = String.Format("ToImage-img-{0}.png", i);

                // Convert the current page to an image with a resolution of 300x300 pixels.
                using (Image image = doc.SaveAsImage(i, 300, 300))
                {
                    // Save the image as a PNG file.
                    image.Save(fileName, System.Drawing.Imaging.ImageFormat.Png);

                    // Open the saved image file using the default associated program.
                    System.Diagnostics.Process.Start(fileName);
                }
            }

            // Close the PDF document.
            doc.Close();
        }
    }
}

File: VB/13_Conversion/ToImage.vb
Imports System.Drawing.Imaging
Imports Spire.Pdf

Namespace ToImage
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Specify the file path of the PDF document to be loaded
            Dim file As String = "..\..\..\..\..\..\Data\ToImage.pdf"

            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Load the PDF document from the specified file path
            doc.LoadFromFile(file)

            ' Save each page of the PDF document as an image
            For i As Integer = 0 To doc.Pages.Count - 1
                ' Generate a unique file name for each image
                Dim fileName As String = String.Format("ToImage-img-{0}.png", i)

                ' Convert the page to an image and save it
                Using image As Image = doc.SaveAsImage(i, 300, 300)
                    image.Save(fileName, ImageFormat.Png)

                    ' Open the saved image using the default associated application
                    Process.Start(fileName)
                End Using
                ' =============================================================================
                ' Use the following code for netstandard dlls
                ' =============================================================================
                'Using image = doc.SaveAsImage(i)
                '    Dim filename As String = String.Format(outputFile & i & ".bmp")
                '    Using fileStream As New System.IO.FileStream(filename, System.IO.FileMode.Create, System.IO.FileAccess.Write)
                '        image.CopyTo(fileStream)
                '        fileStream.Flush()
                '    End Using
                'End Using
                ' =============================================================================

            Next i

            ' Close the PDF document
            doc.Close()
        End Sub
    End Class
End Namespace
